### PR TITLE
feat: add settings toggle for simulations

### PR DESCRIPTION
**Pull request by OGPoyraz**
## **Description**

This PR aims to add simulations toggle(`Settings > Security&Privacy > Transactions > Estimate Balance Changes`) and adds simulation notification to "what's new" modal. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23604?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2241

## **Manual testing steps**

Since simulations are hidden behind code-fence, you need to enable by putting `- transaction-simulation` in to `builds.yml#22` then you should start extension as usual `yarn start`

- When you first time enable and open extension, "what's new" modal should show simulations section.
- Toggle should appear under Settings > Security&Privacy > Transactions > Estimate Balance Changes

## **Screenshots/Recordings**

Screenshot 2024-03-20 at 11 29 43
Screenshot 2024-03-20 at 11 30 34

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/_locales/en/messages.json [hunk 1]
```diff
@@ -3106,6 +3106,12 @@
   "notificationsPortfolioV2Title": {
     "message": "MetaMask Portfolio - join millions"
   },
+  "notificationsSimulationsDescriptionOne": {
+    "message": "Now you can see the potential outcome of your transactions before you make them!"
+  },
+  "notificationsSimulationsDescriptionTwo": {
+    "message": "This is just a simulation, so we can’t guarantee the final outcome. You can turn this off anytime in Settings > Security & Privacy. "
```
Comment (matthewwalsh0) [inline app/_locales/en/messages.json:3113]: Do we need a line break after `anytime in` to ensure the settings path get it's own line and doesn't wrap?

File: app/scripts/controllers/preferences.js [hunk 2]
```diff
@@ -111,6 +111,9 @@ export default class PreferencesController {
       ///: END:ONLY_INCLUDE_IF
       isLineaMainnetReleased: false,
       useExternalNameSources: true,
+      ///: BEGIN:ONLY_INCLUDE_IF(transaction-simulation)
+      useSimulations: true,
```
Comment (matthewwalsh0) [inline app/scripts/controllers/preferences.js:115]: Just for future proofing, should we be specific and call this `useTransactionSimulations`?

File: ui/store/actions.ts [hunk 3]
```diff
@@ -4741,6 +4741,16 @@ export function setUseExternalNameSources(val: boolean): void {
   }
 }
 
+///: BEGIN:ONLY_INCLUDE_IF(transaction-simulation)
+export function setUseSimulations(val: boolean): void {
```
Comment (matthewwalsh0) [inline ui/store/actions.ts:4745]: Again for clarity, should we rename all the various methods and actions to `TransactionSimulations`?

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 4]
```diff
@@ -907,6 +911,43 @@ export default class SecurityTab extends PureComponent {
     );
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IF(transaction-simulation)
+  renderSimulationsToggle() {
+    const { t } = this.context;
+    const { useSimulations, setUseSimulations } = this.props;
+
+    return (
+      <Box
```
Comment (matthewwalsh0) [inline ui/pages/settings/security-tab/security-tab.component.js:920]: Each time we add one of these, I wish all this duplication was in a `SecurityToggleButton` function to massively shrink this file.

But as usual, not worth the time or risk at this point 😄

File: ui/pages/settings/security-tab/security-tab.component.js [hunk 5]
```diff
@@ -968,6 +1009,9 @@ export default class SecurityTab extends PureComponent {
         <div className="settings-page__content-padded">
           {this.renderCurrencyRateCheckToggle()}
           {this.renderIncomingTransactionsOptIn()}
+          {/* ///: BEGIN:ONLY_INCLUDE_IF(transaction-simulation) */}
+          {this.renderSimulationsToggle()}
```
Comment (matthewwalsh0) [inline ui/pages/settings/security-tab/security-tab.component.js:1013]: In addition to the privacy settings, I think Barbara requested we add this to the onboarding settings also.

Which could be in: `ui/pages/onboarding-flow/privacy-settings/privacy-settings.js`?

Comment (OGPoyraz) [reply ui/pages/settings/security-tab/security-tab.component.js:1014]: It totally slipped my mind after implementing onboarding pages, added that in https://github.com/MetaMask/metamask-extension/pull/23604/commits/c775aae0988b548cfd88d771e552801270597613

File: ui/pages/settings/security-tab/security-tab.component.js (hunk 4 above)
Comment (OGPoyraz) [reply ui/pages/settings/security-tab/security-tab.component.js:921]: Exactly Matthew, I was like amazed all those repetition
It's been fixed in the onboarding privacy settings, might touch this later :)

File: ui/store/actions.ts (hunk 3 above)
Comment (OGPoyraz) [reply ui/store/actions.ts]: Renamed in https://github.com/MetaMask/metamask-extension/pull/23604/commits/c775aae0988b548cfd88d771e552801270597613

File: app/_locales/en/messages.json (hunk 1 above)
Comment (OGPoyraz) [reply app/_locales/en/messages.json]: Good call, I've added a new line after `anytime in`

File: app/_locales/en/messages.json [hunk 6]
```diff
@@ -3115,6 +3115,12 @@
   "notificationsPortfolioV2Title": {
     "message": "MetaMask Portfolio - join millions"
   },
+  "notificationsSimulationsDescriptionOne": {
+    "message": "Now you can see the potential outcome of your transactions before you make them!"
+  },
+  "notificationsSimulationsDescriptionTwo": {
+    "message": "This is just a simulation, so we can’t guarantee the final outcome. You can turn this off anytime in \nSettings > Security & Privacy. "
```
Comment (vinistevam) [inline app/_locales/en/messages.json:3122]: Is `\n` necessary before `Settings > Security`?

**Comment by vinistevam** [comment]
Great PR @OGPoyraz !!!
Is this PR also including this toggle in the search field in the Settings? If so, I think we need to add in [settings.js](https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/constants/settings.js). Otherwise happy to approve.

File: app/_locales/en/messages.json (hunk 6 above)
Comment (OGPoyraz) [reply app/_locales/en/messages.json:3122]: We did it intentionally to make that route visible in the new line.

https://github.com/MetaMask/metamask-extension/pull/23604#discussion_r1532799079

**Comment by OGPoyraz** [comment]
> Great PR @OGPoyraz !!! Is this PR also including this toggle in the search field in the Settings? If so, I think we need to add in [settings.js](https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/constants/settings.js). Otherwise happy to approve.

Good call Vini it's now added.

**Review by vinistevam — DISMISSED** [review]

File: ui/helpers/constants/settings.js [hunk 7]
```diff
@@ -287,6 +287,16 @@ export const SETTINGS_CONSTANTS = [
     route: `${SECURITY_ROUTE}#security-alerts-opensea`,
     icon: 'fa fa-lock',
   },
+  ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
```
Comment (matthewwalsh0) [inline ui/helpers/constants/settings.js:290]: Wrong code fence?

Comment (OGPoyraz) [reply ui/helpers/constants/settings.js:290]: busteeeed

Comment (OGPoyraz) [reply ui/helpers/constants/settings.js]: Co-pilot was not doing his part, sorry :D

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by vinistevam — DISMISSED** [review]

**Review by vinistevam — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Comment by sleepytanya** [comment]
Screenshot 2024-03-22 at 10 22 28 AM
Screenshot 2024-03-21 at 3 09 41 PM

**Pull request merged**